lib/improver/cli/threshold.py

```diff
@@ -193,64 +78,64 @@ def process(cube, threshold_values=None, threshold_dict=None,
+            search for an occurrence
 
     Returns:
         iris.cube.Cube:
-            processed Cube.
+            Cube of probabilities relative to the given thresholds
 
     Raises:
-        RuntimeError:
-            If threshold_dict and threshold_values are both used.
+        ValueError: If threshold_config and threshold_values are both set
+        ValueError: If threshold_config is used for fuzzy thresholding
 
      Warns:
-        warning:
-            If collapsing coordinates with a masked array.
+        UserWarning: If collapsing coordinates with a masked array
 
     """
-    if threshold_dict and threshold_values:
-        raise RuntimeError('threshold_dict cannot be used '
-                           'with threshold_values')
-    if threshold_dict:
-        try:
-            thresholds = []
-            fuzzy_bounds = []
-            is_fuzzy = True
-            for key in threshold_dict.keys():
-                thresholds.append(float(key))
-                if is_fuzzy:
-                    # If the first threshold has no bounds, fuzzy_bounds is
-                    # set to None and subsequent bounds checks are skipped
-                    if threshold_dict[key] == "None":
-                        is_fuzzy = False
-                        fuzzy_bounds = None
-                    else:
-                        fuzzy_bounds.append(tuple(threshold_dict[key]))
-        except ValueError as err:
-            # Extend error message with hint for common JSON error.
-            raise type(err)(
-                "{} in threshold dictionary file. \nHINT: Try adding a zero "
-                "after the decimal point.".format(err))
-        except Exception as err:
-            # Extend any errors with message about WHERE this occurred.
-            raise type(err)("{} in dictionary file.".format(err))
+    import warnings
+    import numpy as np
+
+    from improver.blending.calculate_weights_and_blend import WeightAndBlend
+    from improver.metadata.probabilistic import in_vicinity_name_format
+    from improver.threshold import BasicThreshold
+    from improver.utilities.spatial import OccurrenceWithinVicinity
+
+    if threshold_config and threshold_values:
+        raise ValueError(
+            "--threshold-config and --threshold-values are mutually exclusive "
+            "- please set one or the other, not both")
+    if threshold_config and fuzzy_factor:
+        raise ValueError(
+            "--threshold-config cannot be used for fuzzy thresholding")
+
+    if threshold_config:
+        thresholds = []
+        fuzzy_bounds = []
+        for key in threshold_config.keys():
+            thresholds.append(np.float32(key))
+            # If the first threshold has no bounds, fuzzy_bounds is
+            # set to None and subsequent bounds checks are skipped
+            if threshold_config[key] == "None":
+                fuzzy_bounds = None
+                break
```

Comment:
```suggestion
                continue
```
@cgsandford for info